src/main/webapp/linkWizard/LinkWizard.js

```diff
@@ -0,0 +1,416 @@
+  }
+
+  getAllSampleCollections = () => {
+    SampleCollections.getSampleCollections(this.props.sampleSearchUrl).then(
+      resp => {
+        const sampleCollectionsList = resp.data.map(item => {
+          return {
+            key: item.id,
+            value: item.collectionId,
+            label: item.collectionId + ": " + item.name + " ( " + item.category + " )"
+          };
+        });
+        this.setState({ sampleCollectionList: sampleCollectionsList })
+      }
+    );
+  };
+
+  getConsentGroups = () => {
+    ConsentGroup.getConsentGroupNames(this.props.getConsentGroups).then(
+      resp => {
+        const existingConsentGroups = resp.data.map(item => {
+          return {
+            key: item.id,
+            value: item.label,
+            label: item.label
+          }
+        });
+        this.setState({ existingConsentGroups: existingConsentGroups });
+      });
+  };
+
+  getUserSession() {
+    User.getUserSession(this.props.getUserUrl).then(
+      resp => this.setState({ user: resp.data })
+    )
+  }
+
+  initDocuments() {
+    let documents = [];
+    DOCUMENT_TYPE.forEach(type => {
+      documents.push({ value: type, label: type });
+    });
+
+    this.setState({
+      documentOptions: documents
+    });
+  }
+
+  changeSubmitState = () => {
+    this.setState(prev => {
+      prev.formSubmitted = !prev.formSubmitted;
+      return prev;
+    });
+  };
+
+  toggleSubmitError = () => {
+    this.setState(prev => {
+      prev.submitError = true;
+      return prev;
+    });
+  };
+
+  determinationHandler = (determination) => {
+    this.setState(prev => {
+      prev.determination = determination;
+      if (this.state.determination.projectType !== null && this.state.showInternationalCohortsError === true) {
+        prev.showInternationalCohortsError = false;
+      }
+      return prev;
+    }, () => {
+      this.isValid(null);
+    })
+  };
+
+  updateInfoSecurityFormData = (updatedForm) => {
+    this.setState(prev => {
+      prev.securityInfoFormData = updatedForm;
+      return prev;
+    }, () => {
+      this.isValid(null);
+    })
+  };
+
+  handleInfoSecurityValidity = (isValid) => {
+    this.setState(prev => {
+      prev.isInfoSecurityValid = isValid;
+      prev.showErrorInfoSecurity = !isValid;
+      return prev;
+    })
+  };
+
+  validateMTA() {
+    let isValid = true;
+    if (this.state.linkFormData.requireMta === undefined || this.state.linkFormData.requireMta === '') {
+      isValid = false;
+    }
+    this.setState(prev => {
+      prev.errors.requireMta = !isValid;
+      return prev;
+    });
+    return isValid
+  }
+
+  updateMTA = (updatedForm, field) => {
+    this.setState(prev => {
+      prev.linkFormData = updatedForm;
+      return prev;
+    }, () => {
+      this.isValid(field);
+    })
+  };
+
+  componentDidCatch(error, info) {
+    console.log('----------------------- new consent group error ----------------------');
+    console.log(error, info);
```

Comment:
Have we agreed on this pattern (and if so, I think it's a mistake to keep doing this)? I would prefer that we don't log errors this way and instead do something smarter, like inform the user that an error happened.

Reply:
We have been using this pattern since we started working with React in ORSP. This has been replicated in every new component. I agree is not the best one, we just have not prioritized, defined or discussed a better, generic approach to handle errors. We should define how / how much info to include in an error page. Surly add a story to improve this.

Reply:
Thank you!